File: hanoi.go
package main

import "fmt"

// n개의 원판을 start 에서 end 로 옮기는데, 중간 거치 막대로 mid 를 사용함.
func hanoi(n int, start int, mid int, end int) {
	if n == 1 {
		fmt.Println(start, "->", end)
		return
	}
	hanoi(n-1, start, end, mid)
	hanoi(1, start, mid, end)
	hanoi(n-1, mid, start, end)
}
func main() {
	hanoi(3, 1, 3, 2)
}

File: hanoi.py

# coding: utf-8

# In[1]:

print("Hello Python!")


# In[2]:

print("Hello 역량 강화")


# In[7]:

def hanoi(n, start, end, mid):
    if n == 1:
        print(start, "->", end)
        return
    
    hanoi(n-1, start, mid, end)
    hanoi(1, start, end, mid)
    hanoi(n-1, mid, end, start)


# In[8]:

hanoi(3, 1, 2, 3)


# In[ ]:



